instruments/rigol/rigoldp832.py

```diff
@@ -0,0 +1,344 @@
+    +==============+====================+
+    | DP831A/DP831 | 3                  |
+    +--------------+--------------------+
+    | DP832A/DP832 | 3                  |
+    +--------------+--------------------+
+    | DP821A/DP821 | 2                  |
+    +--------------+--------------------+
+    | DP811A/DP811 | 1                  |
+    +--------------+--------------------+
+    
+    Example usage:
+
+    >>> import instruments as ik
+    >>> psu = ik.rigol.rigoldp832.open_gpibusb('/dev/ttyUSB0', 1)
+    >>> psu.channel[0].voltage = 10 # Sets channel 1 voltage to 10 V.
+
+    The user manual can be found here:
+    http://beyondmeasure.rigoltech.com/acton/attachment/1579/f-03a1/1/-/-/-/-/DP800%20Programming%20Guide.pdf
+    """
+
+    def __init__(self, filelike):
+        super(RigolDP832, self).__init__(filelike)
+        self._channel_count = 3
+
+    # ENUMS #
+
+    class Mode(Enum):
+        """
+        Enum containing valid output modes for DP832
+        """
+        constant_curr = "CC"
+        cont_volt     = "CV"
+        unreg         = "UR"
+
+    # INNER CLASSES #
+
+    class Channel(PowerSupplyChannel):
+        """
+        Class representing a power supply channel on the DP832
+
+        .. warning:: This class should NOT be manually created by the user. It is
+        designed to be initialized by the `DP832` class.
+        """
+
+        def __init__(self, dp, idx):
+            self._dp  = dp
+            self._idx = idx + 1
+            
+        # COMMUNICATION METHODS #
+
+        def _format_cmd(self, cmd):
+            
+            cmd = cmd.split("|") # Arbitrary data delimiter
+            assert len(cmd) <= 2 # Assumption to check.
+
+            cmd = cmd[0].format(idx=self._idx, value=cmd[1])
+            
+            return cmd
+
+        def sendcmd(self, cmd):
+            """
+            Function used to send a command to the instrument while wrapping
+            the command with the neccessary identifier for the channel.
+            :param str cmd: Command that will be sent to the instrument after
+                being prefixed with the channel identifier
+            """
+            cmd = self._format_cmd(cmd)
+            self._dp.sendcmd(cmd)
+
+        def query(self, cmd):
+            """
+            Function used to send a command to the instrument while wrapping
+            the command with the neccessary identifier for the channel.
+            :param str cmd: Command that will be sent to the instrument after
+                being prefixed with the channel identifier
+            :return: The result from the query
+            :rtype: `str`
+            """
+            cmd = self._format_cmd(cmd)
+            return self._dp.query(cmd)
+
+        # PROPERTIES #
+
+        @property
+        def mode(self):
+            """
+            Gets the mode for the specified channel.
+            """
+            return RigolDP832.Mode(self.query("OUTP:MODE? CH{idx}"))
```

Comment:
Just as an example here, I am wondering if the following might be more straight forward:
```python
return RigolDP832.Mode(self.query(f"OUTP:MODE? CH{self._idx}"))
```
This way you can avoid the routine to format strings completely and it doesn't make the sent command all that more complicated, no? Just an idea :)